### PR TITLE
perf(swc_common):split-hygine-data

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
**Description:**
1. Split `HygineData` into two parts. `HygineData.maks` and (`HygineData.syntax_contexts`, `HygineData.markings`), are always used exclusively, using two `Mutex` could reduce some unnecessary locking. e.g. It is not necessary to lock  (`HygineData.syntax_contexts`, `HygineData.markings`) when invoke `Mark::refresh() `, vise versa.
This give us about 10% performance improvement in `transform_base/parallel`
**before**: 
```
es/base/parallel/resolver/typescript                                                                          
                        time:   [13.508 s 13.631 s 13.743 s]
Found 1 outliers among 10 measurements (10.00%)
```

**after**:
```
es/base/parallel/resolver/typescript                                                                          
                        time:   [12.107 s 12.247 s 12.382 s]
```

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "eslint.enable": false,
     "cSpell.allowCompoundWords": true,
     "cSpell.caseSensitive": true,
-    "rust-analyzer.checkOnSave.command": "clippy",
+    // "rust-analyzer.checkOnSave.command": "clippy",
```
Comment (kdy1) [inline .vscode/settings.json:28]: Please revert this

File: crates/swc_common/src/syntax_pos/hygiene.rs [hunk 2]
```diff
@@ -316,6 +309,16 @@ impl HygieneData {
     }
 }
 
+pub fn with_marks<T, F: FnOnce(&mut Vec<MarkData>) -> T>(f: F) -> T {
```
Comment (kdy1) [inline crates/swc_common/src/syntax_pos/hygiene.rs:312]: Please use `pub(crate)`

Comment (IWANABETHATGUY) [reply crates/swc_common/src/syntax_pos/hygiene.rs]: Done

File: crates/swc_common/src/syntax_pos/hygiene.rs [hunk 3]
```diff
@@ -63,9 +63,9 @@ pub struct Mark(u32);
 
 #[allow(unused)]
 #[derive(Clone, Debug)]
-struct MarkData {
-    parent: Mark,
-    is_builtin: bool,
+pub struct MarkData {
```
Comment (kdy1) [inline crates/swc_common/src/syntax_pos/hygiene.rs:66]: Please reduce the scope

Comment (kdy1) [reply crates/swc_common/src/syntax_pos/hygiene.rs:66]: In all places, not only this

Comment (IWANABETHATGUY) [reply crates/swc_common/src/syntax_pos/hygiene.rs]: Done

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_common

**Pull request merged**